src/base/path.cpp

```diff
@@ -155,7 +155,9 @@ QString Path::extension() const
 
 bool Path::hasExtension(const QString &ext) const
 {
-    return (extension().compare(ext, Qt::CaseInsensitive) == 0);
+    Q_ASSERT(ext.startsWith(QLatin1Char('.')));
```

Comment:
I would add another condition: `ext.length() >= 2`

Reply:
> I would add another condition: `ext.length() >= 2`

Then we should modify `Path::extension()` counterpart to be consistent and never return `.` as extension, shouldn't we?

Reply:
>Then we should modify Path::extension() counterpart to be consistent and never return . as extension, shouldn't we?

Are you implying `.` itself is an extension? is it?

Reply:
>Are you implying . itself is an extension? is it?

https://en.wikipedia.org/wiki/Filename_extension mentioned it will depend on filesystem. I tested, linux treats `abc.` as no file extension and windows explorer will trim `abc.` to `abc` automatically when renaming and I reckon it treats `.` itself as part of file extension but it don't like it being empty. Given the above, I think `ext.length() >= 2` is not unexpected.

Reply:
`std::filesystem::path` treats `abc.` as file with `.` extension. But I really dislike some aspects of `std::filesystem::path` including this one.

> Given the above, I think `ext.length() >= 2` condition can still be expected.

I agree. I just want all the extension-related methods to be consistent.